### PR TITLE
Allow adding qp to virtual dc infotons

### PR DESCRIPTION
**Pull request by e-orz**
Adding &qp=... to /proc/dc/someId is now allowed.

This is part of adding dc-sync to ability to sync only part of the data. The virtual infoton is used by the monitor and also in dc itself to check what is the last infoton of the desired query.

### CODE REVIEW

File: server/cmwell-fts/src/main/scala/cmwell/fts/FTSServiceES.scala [hunk 1]
```diff
@@ -1050,9 +1050,9 @@ class FTSServiceES private(classPathConfigFile: String, waitForGreen: Boolean)
       None,
       Some(MultiFieldFilter(Must, Seq(
         SingleFieldFilter(Must, Equals, "system.dc", Some(dc)),                                 //ONLY DC
-        SingleFieldFilter(MustNot, Contains, "system.parent.parent_hierarchy", Some("/meta/")), //NO META
-        SingleFieldFilter(Must, GreaterThan, "system.lastModified", Some("1970"))
-      ))),
+        SingleFieldFilter(MustNot, Contains, "system.parent.parent_hierarchy", Some("/meta/")) //NO META
+      ) ++ fieldFilters
```
Comment (hochgi) [inline server/cmwell-fts/src/main/scala/cmwell/fts/FTSServiceES.scala]: I think it's best to break down the statement to smaller parts. e.g:
```
val filtersSeq: List[FieldFilter] = ...
Some(MultiFieldFilter(Must,
                      fieldFilters.fold(filtersSeq)(filtersSeq.::))) // same as (_ :: filtersSeq) 
```

File: server/cmwell-fts/src/main/scala/cmwell/fts/FTSServiceNew.scala [hunk 2]
```diff
@@ -1006,9 +1006,9 @@ class FTSServiceNew(config: Config, esClasspathYaml: String) extends FTSServiceO
       None,
       Some(MultiFieldFilter(Must, Seq(
         SingleFieldFilter(Must, Equals, "system.dc", Some(dc)),                                 //ONLY DC
-        SingleFieldFilter(MustNot, Contains, "system.parent.parent_hierarchy", Some("/meta/")), //NO META
-        SingleFieldFilter(Must, GreaterThan, "system.lastModified", Some("1970"))
-      ))),
+        SingleFieldFilter(MustNot, Contains, "system.parent.parent_hierarchy", Some("/meta/")) //NO META
+      ) ++ fieldFilters
```
Comment (hochgi) [inline server/cmwell-fts/src/main/scala/cmwell/fts/FTSServiceNew.scala]: same

File: server/cmwell-ws/app/controllers/Application.scala [hunk 3]
```diff
@@ -311,20 +311,23 @@ callback=< [URL] >
...
-    //    infotonOptionToReply(req,ActiveInfotonGenerator.generateInfoton(req.host, path, new DateTime(),length,offset))
-
-    activeInfotonGenerator
-      .generateInfoton(req.host, path, new DateTime(), length, offset, isRoot, nbg)
-      .flatMap(iOpt => infotonOptionToReply(req, iOpt.map(VirtualInfoton.v2i)))
-  }.recover(asyncErrorHandler).get
+  private def handleGetForActiveInfoton(req: Request[AnyContent], path: String) =
+    getQueryString("qp")(req.queryString)
+      .fold(Success(None): Try[Option[RawFieldFilter]])(FieldFilterParser.parseQueryParams(_).map(Some.apply))
+      .map { qpOpt =>
```
Comment (hochgi) [inline server/cmwell-ws/app/controllers/Application.scala]: dangerous. use `flatMap{ qpOpt => Try { ...`

Comment (e-orz) [reply server/cmwell-ws/app/controllers/Application.scala]: I understand. but...
It's the same pattern as `handleSearch`. Is this a mistake also there?

Comment (hochgi) [reply server/cmwell-ws/app/controllers/Application.scala]: Not nessecarily. `authUtils.isValidatedAs` may fail (e.g: unable to retrieve use infoton), but is not called from search...

Though, I won't argue against being more defensive there too.

Comment (hochgi) [reply server/cmwell-ws/app/controllers/Application.scala]: taking it back... it's actually OK. my bad...

**Pull request merged**